web/src/lib/components/user-settings-page/partner-settings.svelte

```diff
@@ -0,0 +1,99 @@
+<script lang="ts">
+	import { UserResponseDto, api } from '@api';
+	import CircleAvatar from '../shared-components/circle-avatar.svelte';
+	import Close from 'svelte-material-icons/Close.svelte';
+	import Button from '../elements/buttons/button.svelte';
+	import PartnerSelectionModal from './partner-selection-modal.svelte';
+	import { handleError } from '../../utils/handle-error';
+	import { browser } from '$app/environment';
+	import { onMount } from 'svelte';
+
+	let partners: UserResponseDto[] = [];
+	let showAddUserModal = false;
+
+	$: {
+		if (browser) {
+			if (showAddUserModal) {
+				document.body.style.overflow = 'hidden';
+			} else {
+				document.body.style.overflow = 'auto';
+			}
+		}
+	}
+
+	const handleAddPartner = async () => {
+		try {
+			showAddUserModal = true;
+		} catch (error) {
+			handleError(error, 'Unable to add partner');
+		}
+	};
+
+	const handleRemovePartner = async (user: UserResponseDto) => {
+		try {
+			await api.partnerApi.removePartner(user.id);
+			partners = await loadPartners();
+		} catch (error) {
+			handleError(error, 'Unable to remove partner');
+		}
+	};
+
+	const addUserHandler = async (event: CustomEvent) => {
+		const { selectedUsers }: { selectedUsers: UserResponseDto[] } = event.detail;
+
+		try {
+			for (const user of selectedUsers) {
+				await api.partnerApi.addPartner(user.id);
+			}
+
+			partners = await loadPartners();
+			showAddUserModal = false;
+		} catch (error) {
+			handleError(error, 'Unable to add partners');
+		}
+	};
+
+	const loadPartners = async () => {
+		const { data: partners } = await api.partnerApi.getPartners('shared-by');
+		return partners.map((partner) => partner.sharedWith);
+	};
+
+	onMount(async () => {
+		partners = await loadPartners();
+	});
+</script>
+
+<section class="my-4">
+	{#if partners.length > 0}
+		<div class="flex flex-row gap-4">
+			{#each partners as partner}
+				<div class="flex rounded-lg gap-4 py-4 px-5 transition-all">
+					<CircleAvatar user={partner} />
```

Comment:
The `CircleAvatar` is a button and clickable, but there's no action associated with clicking it

Reply:
Hm, in the future I believe it should be the action to show sharing settings (dialog or even separate page) for the partner. But now I can imagine only remove partner action. Not sure it is better than no action. 

What do you think?